Move external modules to standalone repos using west

nashif opened the pull request:
This moves the following:

- open-amp
- libmetal
- cypress hal
- silabs hal

nashif committed:
cypress: move HAL to external repo

This HAL is now in its own repository on
https://github.com/zephyrproject-rtos/hal_cypress

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
silabs: move HAL to external repo

This HAL is now in its own repository on
https://github.com/zephyrproject-rtos/hal_silabs

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ext: move libmetal to an external module

Move libmetal to be an external module and add it manifest.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ext: move open-amp to an external module

Move open-amp to be an external module and add it manifest.

See https://github.com/zephyrproject-rtos/open-amp

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

mbolivar commented on west.yml:
```diff
@@ -47,12 +47,24 @@ manifest:
       revision: df96914dd989907f3a5de4cb95b116c2f305820d
       path: modules/fs/fatfs
+    - name: hal_cypress
+      revision: master
```
Please set the revision for `hal_cypress` and `hal_silabs` to a tag or SHA, to preserve bisectability.

nashif replied:
ah, good catch, missed those..

nashif committed:
modules: move kconfig into tree

Move Kconfig from modules to the main zephyr tree to allow depenedency
resolution.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
